From 07f75786b7ec9c4aa4731d85471fd231a46cfd93 Mon Sep 17 00:00:00 2001
From: Julian Greilich <j.greilich@gmx.de>
Date: Wed, 3 Sep 2025 14:30:12 +0200
Subject: a11y: Enable the ValueInterface for the ProgressBar

Enabling the ValueInterface is necessary to use that information
for a better a11y support of the ProgressBar.

Task-number: QTBUG-139712
Pick-to: 6.8 6.9 6.10
Change-Id: Ia5e6c1ba4f6cada8e02b7ed4123454302b494b5d
Reviewed-by: Volker Hilsheimer <volker.hilsheimer@qt.io>
(cherry picked from commit 775b4abc5f5b676ae377fd47adfe4712e7a54e18)
---
 src/quick/accessible/qaccessiblequickitem.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git x/qtdeclarative/src/quick/accessible/qaccessiblequickitem.cpp y/qtdeclarative/src/quick/accessible/qaccessiblequickitem.cpp
index c0f6b42425198af7aac30040c28d4e1a230c96c6..88e0b25ca51af9f0c99b945c91c034cb5ad09559 100644
--- x/qtdeclarative/src/quick/accessible/qaccessiblequickitem.cpp
+++ y/qtdeclarative/src/quick/accessible/qaccessiblequickitem.cpp
@@ -688,7 +688,8 @@ void *QAccessibleQuickItem::interface_cast(QAccessible::InterfaceType t)
            (r == QAccessible::Slider ||
             r == QAccessible::SpinBox ||
             r == QAccessible::Dial ||
-            r == QAccessible::ScrollBar))
+            r == QAccessible::ScrollBar ||
+            r == QAccessible::ProgressBar))
        return static_cast<QAccessibleValueInterface*>(this);
 
     if (t == QAccessible::TextInterface) {
